### DISCUSSION TITLE
Multiple Save Paths in a single game manifest

### DISCUSSION THREAD

**Discussion by redactedscribe** [MikeMaximus/gbm#281] [Ideas]
In some cases, game save data and game configuration files aren't stored under the same root directory, leading to a need to create two manifests to cover backing up both bits of data (which I argue should be the base requirement to consider a game's user data "backed up").

I've observed that some manifests backup an entire directory and make no effort to exclude user settings found within the path. Others, such as those found via Ludusavi, sometimes have two separate manifests where the first backs up both the game save data _and_ the user configuration files, and the second backs up only the subdirectory with the config files. This dual manifest approach seem to be just a means to deal with the PCGamingWiki game articles and make use of paths found on there.

Looking at the Official list of manifests, it looks like the original goal of the program was to solely backup save game files as we see only 1-manifest-per-game: if configuration files happen to be in the same path, great, an official manifest will back those up too, if not, forget about them.

Ideally GBM could stick to its 1-manifest-per-game target and not start to split manifests into data and settings as is frequent with Ludusavi imports. With a change to supporting multiple Save Paths in a single manifest, I assume even Ludusavi manifest imports could be merged into just 1-manifest-per-game too. If someone wants the save data split from the settings files, they can create two separate manifests to handle each separately, but I don't think it should be the default approach for solving the problem.

I presume the 1-manifest-per-game was that way from the start since the core of the program is a UI to pass arguments to 7-Zip and that limits the options the program can do based on the command line flags 7-Zip provides. But I think multiple Save Paths could still work.

If each Save Path had its own include and exclude parameters, then like individual manifests, one archive per Save Path could be created using compression, and then for convenience, both of those could be stored (not compressed) into another archive (or just a plain folder - maybe a choice for the Settings).

The main benefits would be:

- Make Official manifests have the capability to intentionally save user config files (without the need to create multiple manifests).
- Multiple Ludusavi manifests for the same game could be merged into single manifests.
- Save Paths could be designed so that they could be disabled, and so if you wouldn't ever want the manifest backing up the Save Path which handles config files, just uncheck it.
- A purely 1-manifest-per-game system would mean there shouldn't be manifests with duplicate names requiring UUIDs in subdirectory names etc, and should help simplify some matters internally and for the user.
- ...?

Would making a strictly 1-manifest-per-game system really worth the effort?

Thanks.

GBM 1.3.1

**Comment by MikeMaximus** [comment]
Thanks again for the detailed suggestions.

This feature has been requested a few times over the years.  It's very unlikely that I'll be adding support for it to GBM in it's current form.  

GBM is almost a decade old and is going to need to be ported to another platform sooner rather than later.  When that happens, multi-save path configurations per game will be part of the core design.

This issue will be converted to a discussion.
